main/src/cgeo/geocaching/CacheListActivity.java

```diff
@@ -1133,6 +1123,20 @@ private String getDisplayName(final Uri uri) {
     }
 
     public void refreshStored(final List<Geocache> caches) {
+        if (type == CacheListType.OFFLINE && caches.size() > REFRESH_WARNING_THRESHOLD) {
+            Dialogs.confirmYesNo(this, R.string.caches_refresh_all, R.string.caches_refresh_all_warning, new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(final DialogInterface dialog, final int id) {
+                    refreshStoredConfirmed(caches);
+                    dialog.cancel();
```

Comment:
We should keep it the way it is.

I debugged a little bit. 
The next dialog coming up is the ProgressBar with the time estimation. This comes up perfectly on top of the warning dialog.

A problem might arise, when we close the dialog before starting the refresh job. Then the dialog disapears immediately, but now it is possible for an impatient user to click on some cache listing before the progress dialog comes up. This causes the cache listing to open and the progress dialog of the refresh process opens "in the background". He can't see it and can't "cancel" the job as long as the cache listing is on top.